### PR TITLE
Changing State Of Components Outside Your Hierarchy

### PR DESCRIPTION
**Pull request by emarukyan**
*Before* submitting a pull request, please make sure the following is done...

1. Fork the repo and create your branch from `master`.
2. If you've added code that should be tested, add tests!
3. If you've changed APIs, update the documentation.
4. Ensure the test suite passes (`npm test`).
5. Make sure your code lints (`npm run lint`) - we've done our best to make sure these rules match our internal linting guidelines.
6. If you haven't already, complete the [CLA](https://code.facebook.com/cla).

I just added hint for how to change a components state outside of your hierarchy and fire a global action to notify all stakeholders that there is some change, whoever is interested in change, should REACT and do something on their state.
Basically it is the description of Flux, introduced by Facebook.

### CODE REVIEW

**Comment by lacker** [comment]
Thanks for taking the time to send in a pull request. The "Thinking In React" document is sort of a founding React-philosophy document. It's like adding another paragraph to the declaration of independence. I think we should be willing to polish this document to modernize it or clean it up, but rather than fundamentally messing with its structure I think it's better to just create something new. This seems like the sort of useful tip you could submit to http://react.tips/ or find some other place to put it besides the React documentation, so I suggest you look elsewhere. Or you could wait for there to be a "fetching data with React" document which some other open issues / pull requests are referring to.
